x-pack/plugins/security_solution/public/resolver/store/selectors.ts

```diff
@@ -230,15 +240,50 @@ export const relatedEventTotalForProcess = composeSelectors(
  * The bounding box represents what the camera can see. The camera position is a function of time because it can be
  * animated. So in order to get the currently visible entities, we need to pass in time.
  */
-export const visibleProcessNodePositionsAndEdgeLineSegments = createSelector(
-  indexedProcessNodesAndEdgeLineSegments,
-  boundingBox,
-  function (
-    /* eslint-disable no-shadow */
-    indexedProcessNodesAndEdgeLineSegments,
-    boundingBox
-    /* eslint-enable no-shadow */
-  ) {
-    return (time: number) => indexedProcessNodesAndEdgeLineSegments(boundingBox(time));
+export const visibleNodesAndEdgeLines = createSelector(nodesAndEdgelines, boundingBox, function (
+  /* eslint-disable no-shadow */
+  nodesAndEdgelines,
+  boundingBox
+  /* eslint-enable no-shadow */
+) {
+  return (time: number) => nodesAndEdgelines(boundingBox(time));
+});
+
+/**
+ * Takes a nodeID (aka entity_id) and returns the associated aria level as a number or null if the node ID isn't in the tree.
+ */
+export const ariaLevel: (
+  state: ResolverState
+) => (nodeID: string) => number | null = composeSelectors(
+  dataStateSelector,
+  dataSelectors.ariaLevel
+);
+
+/**
+ * Takes a nodeID (aka entity_id) and returns the node ID of the node that aria should 'flowto' or null
+ * If the node has a following sibling that is currently visible, that will be returned, otherwise null.
+ */
+export const ariaFlowtoNodeID: (
+  state: ResolverState
+) => (time: number) => (nodeID: string) => string | null = createSelector(
+  visibleNodesAndEdgeLines,
+  composeSelectors(dataStateSelector, dataSelectors.followingSibling),
+  (visibleNodesAndEdgeLinesAtTime, followingSibling) => {
+    return defaultMemoize((time: number) => {
```

Comment:
clever

Reply:
been thinking about taking advantage of reselect's parametric selectors. siem uses em. /shrug